students/andrew_garcia/lesson_04/Mailroom Part 2.py

```diff
@@ -133,27 +134,29 @@ def sorting_donors(all_donations = all_donations):
         donation_amount = len(all_donations[item])
         total_donations = sum(all_donations[item])
         average_donation = total_donations / donation_amount
-        sorting_donors += [[item, donation_amount, total_donations, average_donation]]
+        sorting_donors.append([item, donation_amount, total_donations, average_donation])
 
     def sort_key(sorting_donors):  # creating the sorting format
         return sorting_donors[2]
 
     # sorts donor by total amount of money donated
     highest_donation = sorted(sorting_donors, key = sort_key)
-    highest_donation = highest_donation[::-1]
+    highest_donation.reverse()
     return highest_donation
 
 
-def create_report(sorting_donors = sorting_donors()):
+def create_report():
     """Creates the report format for donors, sorting by highest donation amount
     Uses the sorting_donors function to have the list of donors in the right order"""
 
 
     print('\nDonor Name          | # Donations |   Average Donation   |   Total Donations  | ')
     print('-------------------------------------------------------------------------------')
-    for item in sorting_donors:
+    sorted_donors = sorting_donors()
+    for item in sorted_donors:
         print(f'{item[0]:27}{item[1]}       ${item[2]:18.2f}    ${item[3]:18.2f}')
-    options_menu()
+
+
 
 if __name__ == '__main__':
     options_menu()
```

Comment:
@ajaggarcia10 - changes look good, but the thank_all function isn't working correctly.  It's creating WAY too many text files.  See if you can fix it and reach out if you're having trouble.